### PR TITLE
Explain neuron models in documentation

### LINKED ISSUE #1866: Explain neuron models in documentation
Currently, it is unclear from the documentation whether or not neuron model features can be combined. 

As discussed in the ONDVC, we should clarify that neuron models in NEST are one entity. The information could be added to the welcome page or models directory.

### PR DESCRIPTION
**Pull request by sarakonradi**
This PR clarifies that neuron models in NEST are one entity. It fixes #1866.

### CODE REVIEW

**Comment by clinssen** [comment]
@sarakonradi: could you also make me a reviewer on this one? Thanks!

**Comment by sarakonradi** [comment]
Done!

File: doc/index.rst [hunk 1]
```diff
@@ -37,8 +37,13 @@ NEST is a simulator for **spiking neural network models**, ideal for networks of
...
 
-**Have an idea of the type of model you need?**
-    Click on one of the images to access our :doc:`model directory <models/index>`:
+**Know which model you need?**
+    You can find a list of all models implemented in NEST in our
+    :doc:`model directory <models/index>`.
+
+    Each neuron model is one entity. Check out
+    `NESTML <https://nestml.readthedocs.io/en/latest/>`_ if you want to combine
+    aspects of different models or create a new model.
```
Comment (jougs) [inline doc/index.rst:46]:
```suggestion
    Please note that each neuron or synapse model available in NEST is a single
    entity. If your research demands a combination of features that are available
    in `different` models, our built-in models may not be for you. In this case you
    should check out `NESTML <https://nestml.readthedocs.io/en/latest/>`_ to
    learn about how to create your own.
```

**Review by jougs — COMMENTED** [review]
Thanks! I've added a slighly longer, but much more detailed version of the main sentence for your consideration.

**Comment by clinssen** [comment]
As this material will be on the front (landing) page of the NEST documentation, I propose to keep it terse and PR-style.

```rst
**Know which model you need?**

NEST comes packaged with a large collection of neuron and synaptic plasticity models. 
You can find a list of all available models in our :doc:`model directory <models/index>`, 
or select a model category by clicking one of the images:

[...]


**Need a different model?**

To customize or combine features of neuron and synapse models, we recommend 
using the `NESTML modeling language <https://nestml.readthedocs.io/>`_.
```

If we want to address the fundamental issue with #1866, we should probably add this information to (and/or restore some of the content of) this document: https://nest.github.io/nest-simulator/neuron_and_device_models. As it's not on the main NEST repository, I can't find the history, could you perhaps a post a link to the right fork/branch?

**Comment by jougs** [comment]
@clinssen: I like your suggestion for the text and would approve a corresponding change in this PR. Regarding the linked page from the developer space, I don't think that it adds anything to your text.

**Review by jougs — APPROVED** [review]
Very nice!

**Comment by jougs** [comment]
@clinssen: what do you think of the current state? Thanks!

**Review by clinssen — APPROVED** [review]
Perfect!

**Pull request merged**